README.md

```diff
@@ -46,11 +46,41 @@ following contents:
   "ConnectionStrings": {}
 }
 ```
-
 ### Starting the functions runtime
 
 ```
 $ yarn start
 ```
 
 The server should reload automatically when the code changes.
+
+### Starting the io-functinos-app docker image
+If you are trying to run the docker images on your local environment (through the docker-compose) you must set the following variables in the `local.settings.json` file:
+  * AzureWebJobsStorage
+  * QueueStorageConnection
+  * AssetsStorageConnection
+With this **connection string** as value:
+  * DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://azurite:10000/devstoreaccount1;QueueEndpoint=http://azurite:10001/devstoreaccount1;TableEndpoint=http://azurite:10002/devstoreaccount1;
+
+The **connection string** has a default value needed to connect to Azurite, a local emulator used to provide a free local environment for testing an Azure Blob, Queue Storage, and Table Storage application.
+As for docker-compose instructions, the Azurite docker image runs the Blob service on port 10000, the Queue service on port 10001 and the Table service on port 10002.
+If Azurite is executed on different address or ports, the **connection string** must be changed according to the service.
+
+These must be the other variables values for the `local.settings.json` file:
+  * COSMOSDB_URI=https://cosmosdb:3000/
+  * COSMOSDB_KEY=dummykey
+  * COSMOSDB_NAME=testdb
+
+The COSMOSDB_URI must be the address of the cosmos db instance specified in the docker-compose.yml file while the COSMOSDB_KEY and COSMOSDB_NAME could be totally randomic.
+
+Then, copy `.env.example` to `.env` and fill the variables with the following mandatory variables:
+ * QueueStorageConnection=**connection string**
```

Comment:
why do we need to set `QueueStorageConnection` bot on local.settings.json and .env?

Reply:
Yes it's a mistake, I'll remove it from the .env.

Reply:
what about putting everything in .env instead and skip local.settings.json? I mean, in order to have a single source of truth for variables. I don't know if it is possibile, just a suggestion.

Reply:
The local.settings.json is used when the io-functions-app project is launched manually from the CLI with the "yarn start" command: it is used as default file for the configuration load by the azure function core tools.
Otherwise the .env file is used when the project is launched as docker container (in the docker-compose file there are 4 services using this file, including the io-functions-app docker image itself) .
For this reason, at the moment, both files can't be deleted.
Maybe to separate the contexts, all the env variables of the io-functions-app can be moved to the local.settings.json (in both case - manually and docker - they are loaded from this file) and the other variables of the services launched as docker containers can stay in the .env file.

Reply:
in which cases do you want to start the project using yarn instead of docker? (and viceversa) maybe we can settle on just one approach?

Reply:
The docker-compose launches locally all the dependencies for the project and the io-functions-app itself using yarn start (but inside a container).
I suppose that running the project using yarn can be useful if you want to connect to remote dependencies (for example directly to the storage and db instances on Azure) without launching all the dependencies on your local machine.
Using one approach, from my point of view, will force the dev to run all the dependencies locally also if io-functions-app will connect to remote instances.

Reply:
in which cases do you want to choose one over the other? in other words: in which scenarios do you want to use docker instead of remote deps and viceversa? can we settle on just one approach or are there edge cases?

Reply:
From my point of view there are two cases:

- **Docker** : Run all the dependencies and the io-functions-app (connecting to these local dependencies) as docker containers -> Test flows and APIs with custom data 
- **CLI** : Run only io-functions-app locally (without the other dependencies) connecting to the Azure remote instances

Reply:
Under which circumstances a developer would prefer the **CLI** approach? 
(I'm not assuming there isn't a reason, just asking why)

Reply:
When the dev wants to reproduce an issue or a flow connecting to the Azure remote instances.
For example, there is a bug on the profile creation.
I'm not able to reproduce it with the local dependencies (as local dependencies we are using azurite that is an emulator of the Azure storage and an alternative version of cosmosdb).
So I start the io-functions-app from CLI configuring the connection strings to the remote Azure instances in order to reproduce the issue with the real dependencies.

Reply:
which kind of issues can arise with real dependencies that cannot be caught using the emulators? can you make an example of a scenario that you cannot reproduce locally? 

moreover: isn't this a responsibility of the DAO layer (ie. for interacting with cosmos)?
while the infrastructure layer (usually encapsulated into SDK packages) should bring integration tests, we can assume that the function integration logic just works, or do you see any other issues?

finally: just in case, can't we run the function inside docker anyway (configuring it using .env) and eventually point the connection strings to the external resources? is there any drawback with this approach?